cranelift/codegen/src/isa/riscv64/inst/args.rs

```diff
@@ -43,19 +50,23 @@ impl AMode {
     }
 
     pub(crate) fn with_allocs(self, allocs: &mut AllocationConsumer<'_>) -> Self {
-        let reg = allocs.next(self.get_base_register());
         match self {
-            AMode::RegOffset(_, offset, ty) => AMode::RegOffset(reg, offset, ty),
-            AMode::SPOffset(..) | AMode::FPOffset(..) | AMode::NominalSPOffset(..) => self,
+            AMode::RegOffset(reg, offset, ty) => AMode::RegOffset(allocs.next(reg), offset, ty),
+            AMode::SPOffset(..) | AMode::FPOffset(..) | AMode::NominalSPOffset(..) => {
+                allocs.next(self.get_base_register().unwrap());
```

Comment:
The calls to `AllocationConsumer::next()` need to match one-to-one with the args provided to RA2; otherwise all the assignments for the following instructions will be shifted.

However, I do agree with your comment below that SP and FP should not be regalloc args in the first place, since they aren't allocatable registers! I had thought we had an assert for this, but maybe not...